lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -1137,34 +1111,30 @@ public function prepare_item_for_response( $post, $request ) {
 		}
 
 		if ( ! empty( $schema['properties']['title'] ) ) {
+			add_filter( 'protected_title_format', array( $this, 'protected_title_format' ) );
 			$data['title'] = array(
 				'raw'      => $post->post_title,
 				'rendered' => get_the_title( $post->ID ),
 			);
+			remove_filter( 'protected_title_format', array( $this, 'protected_title_format' ) );
 		}
 
 		if ( ! empty( $schema['properties']['content'] ) ) {
-
-			if ( ! empty( $post->post_password ) ) {
-				$this->prepare_password_response( $post->post_password );
-			}
-
 			$data['content'] = array(
-				'raw'      => $post->post_content,
+				'raw'       => $post->post_content,
 				/** This filter is documented in wp-includes/post-template.php */
-				'rendered' => apply_filters( 'the_content', $post->post_content ),
+				'rendered'  => post_password_required( $post ) ? '' : apply_filters( 'the_content', $post->post_content ),
+				'protected' => (bool) $post->post_password,
```

Comment:
It's a little counterintuitive to me to have the `protected: true` enumerated in every `content` and `excerpt` object, _vs_ just having it declared on the root level. This feels like it's adding a lot of duplication in to every response object, especially given that `protected: false` is by far the most common case. Why not use a top-level `protected` key?